Wikipedia/Code/UIViewController+WMFOpenExternalUrl.m

```diff
@@ -47,16 +48,26 @@ - (void)wmf_openExternalUrl:(NSURL*)url useSafari:(BOOL)useSafari {
 
 - (void)wmf_openExternalUrlModallyIfNeeded:(NSURL*)url forceSafari:(BOOL)forceSafari {
     // iOS 9 and later just use UIApplication's openURL.
-    if (forceSafari || [[NSProcessInfo processInfo] isOperatingSystemAtLeastVersion:(NSOperatingSystemVersion){9, 0, 0}]) {
+    if (forceSafari) {
         [[UIApplication sharedApplication] openURL:url];
     } else {
         // pre iOS 9 use SVModalWebViewController.
         if (self.presentedViewController) {
             [self dismissViewControllerAnimated:YES completion:^{
-                [self wmf_presentExternalUrlAsSVModal:url];
+                [self wmf_presentExternalUrlWithinApp:url];
             }];
             return;
         }
+        [self wmf_presentExternalUrlWithinApp:url];
+    }
+}
+
+- (void)wmf_presentExternalUrlWithinApp:(NSURL *)url {
+//    iOS 9 and later
+    if ([[NSProcessInfo processInfo] isOperatingSystemAtLeastVersion:(NSOperatingSystemVersion){9, 0, 0}]) {
```

Comment:
I'd suggest using `if ([SFSafariViewController class])` as per the recommendation in https://developer.apple.com/library/ios/documentation/DeveloperTools/Conceptual/cross_development/Using/using.html#//apple_ref/doc/uid/20002000-SW3